index_api-v2.yaml

```diff
@@ -0,0 +1,722 @@
+        - node_id
+      properties:
+        node_id:
+          type: string
+    PostNodeSyncRequest:
+      type: object
+      required:
+        - profile_url
+      properties:
+        profile_url:
+          type: string
+    PostNodeSyncResponse:
+      type: object
+      required:
+        - profile_url
+        - node_id
+        - status
+      properties:
+        profile_url:
+          type: string
+        node_id:
+          type: string
+        status:
+          type: string
+          enum:
+            - received
+            - validated
+            - posted
+            - deleted
+            - validation_failed
+            - post_failed
+        last_updated:
+          type: number
+        profile_hash:
+          type: string
+        failure_reasons:
+          type: array
+          items:
+            type: string
+    GetNodesResponse:
+      type: array
+      items:
+        type: object
+        required:
+          - profile_url
+          - last_updated
+          - linked_schemas
+        properties:
+          profile_url:
+            type: string
+          last_updated:
+            type: number
+          linked_schemas:
+            type: array
+            items:
+              type: string
+          primary_url:
+            type: string
+          geolocation:
+            type: object
+            properties:
+              lat:
+                type: number
+              lon:
+                type: number
+          locality:
+            type: string
+          region:
+            type: string
+          country:
+            type: string
+          status:
+            type: string
+          tags:
+            type: array
+            items:
+              type: string
+    GetNodeStatusResponse:
+      type: object
+      required:
+        - profile_url
+        - node_id
+        - status
+      properties:
+        profile_url:
+          type: string
+        node_id:
+          type: string
+        status:
+          type: string
+          enum:
+            - received
+            - validated
+            - posted
+            - deleted
+            - validation_failed
+            - post_failed
+        last_updated:
+          type: number
+        profile_hash:
+          type: string
+        failure_reasons:
+          type: array
+          items:
+            type: string
+    Error:
+      type: object
+      required:
+        - message
+      properties:
+        message:
+          type: string
+        status:
+          type: integer
+  parameters:
+    node_id:
+      name: node_id
+      in: path
+      description: The ID of the node
+      required: true
+      schema:
+        type: string
+    schema:
+      name: schema
+      in: query
+      description: The name of the schema
+      schema:
+        type: string
+    last_updated:
+      name: last_updated
+      in: query
+      description: Unix timestamp in seconds when node was last validated by index
```

Comment:
See comment above re validated vs updated. My assumption is that this will be the (self reported) update time of the node, rather than the last time the index validated the node (for example via a /validate request)

Reply:
See my previous comment about the `/validate` endpoint having no relation to this timestamp.

This timestamp is only updated each time the `POST /nodes` or `POST /nodes-sync` endpoints are sent the `profile_url` of the node in question. If nothing has changed, the timestamp is currently still updated to the moment the Index performed the validation work again. 

I suppose we could make a change that checks if the content of the `profile_url` is exactly the same, and if so, the timestamp remains as it was. Let me know if you think this is worth implementing.

Reply:
I think my comment was a bit unclear in which line it refers to. I was just thinking that the text "Unix timestamp in seconds when node was last validated by index" maybe should be changed to "Unix timestamp in seconds when node was last updated", because the node might get validated by the index without updating this value.

Reply:
I fixed the wording, but my question above still stands. Does it make sense to leave this `last_updated` value unchanged if the `profile_url` is resent to `POST /nodes` (or `POST /nodes-sync`) but it's profile content is unchanged? Currently it the timestamp will be updated regardless.

Reply:
Good question. I think for now it makes sense to leave it as is, rather than comparing the content, but we should think about different scenarios and how this is used.